Mathlib/NumberTheory/ModularForms/JacobiTheta/TwoVariable.lean

```diff
@@ -14,119 +15,509 @@ This file defines the two-variable Jacobi theta function
 $$\theta(z, \tau) = \sum_{n \in \mathbb{Z}} \exp (2 i \pi n z + i \pi n ^ 2 \tau),$$
 
 and proves the functional equation relating the values at `(z, τ)` and `(z / τ, -1 / τ)`,
-using Poisson's summation formula. We also show holomorphy (in both variables individually,
-not jointly).
+using Poisson's summation formula. We also show holomorphy (jointly in both variables).
+
+Additionally, we show some analogous results about the derivative (in the `z`-variable)
+
+$$\theta'(z, τ) = \sum_{n \in \mathbb{Z}} 2 \pi i n \exp (2 i \pi n z + i \pi n ^ 2 \tau).$$
+
+(Note that the Mellin transform of `θ` will give us functional equations for `L`-functions
+of even Dirichlet characters, and that of `θ'` will do the same for odd Dirichlet characters.)
 -/
 
 open Complex Real Asymptotics Filter Topology
 
-open scoped Real BigOperators
+open scoped BigOperators ComplexConjugate
 
-/-- The two-variable Jacobi theta function,
-`θ z τ = ∑' (n : ℤ), cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)`.
+noncomputable section
```

Comment:
I have added
```lean
instance : CharZero ℤ := StrictOrderedSemiring.to_charZero
instance : CharZero ℝ := RCLike.charZero_rclike
instance : Module ℂ ℂ := NormedSpace.toModule
instance : Module ℂ (ℂ × ℂ) := Prod.instModule
instance : AddCommMonoid ℂ := inferInstance
instance : AddCommMonoid (ℂ × ℂ) := inferInstance
instance : TopologicalSpace ℂ := inferInstance
instance : TopologicalSpace (ℂ × ℂ) := inferInstance
instance : SMul ℝ ℂ := Algebra.toSMul
instance : SMul ℂ ℂ := Algebra.toSMul
instance : Module ℝ ℂ := instModule
instance : MulActionWithZero ℂ ℂ := MonoidWithZero.toMulActionWithZero ℂ
instance : MulAction ℂ ℂ := MulActionWithZero.toMulAction
instance : SMulCommClass ℂ ℂ ℂ := smulCommClass_self ℂ ℂ
instance : ContinuousConstSMul ℂ ℂ := SMulCommClass.continuousConstSMul
instance : ContinuousConstSMul ℝ ℂ :=
  UniformContinuousConstSMul.to_continuousConstSMul ℝ ℂ
instance : IsScalarTower ℂ ℂ ℂ := IsScalarTower.right
instance : IsScalarTower ℝ ℂ ℂ := IsScalarTower.complexToReal
instance : SMulCommClass ℂ ℝ ℂ := IsScalarTower.to_smulCommClass'
instance : AddCommGroup (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.addCommGroup
instance : AddCommMonoid (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.addCommMonoid
instance : Module ℝ (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.module
instance : Module ℂ (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.module
instance : SMul ℂ (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.instSMul
instance : NormedSpace ℝ (ℂ × ℂ →L[ℂ] ℂ) := ContinuousLinearMap.toNormedSpace
instance : FiniteDimensional ℂ (ℂ × ℂ →L[ℂ] ℂ) := inferInstance
instance : FiniteDimensional ℝ (ℂ × ℂ →L[ℂ] ℂ) := FiniteDimensional.complexToReal (ℂ × ℂ →L[ℂ] ℂ)
```
This is certainly not very nice (and probably some of these are not very important), but it avoids a lot of slow type class synthesis tasks throughout the file.

Generally, it seems like everything involving `ℂ × ℂ →L[ℂ] ℂ` is prone to slow type class inference issues...

Reply:
Does #11980 help here?

Reply:
@riccardobrasca It does help considerably, which is why Michael and I made it a prereq for this PR; but now it's merged.